0003-configuration-management-extension/README.md

```diff
@@ -0,0 +1,93 @@
+---
+name: Configuration Management Extension
+shortname: 3/CMX
+status: raw
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0002-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Multiple configurations available for client._ It is possible that there will be more than one available configurations for endpoint (eg.: endpoint was offline while new configurations were added).
+   Solution:
+   - Only latest configuration should be sent to an endpoint to reduce amount of traffic.
+_Note:_ Approach for FOTA should be different.
+
+2. _The server should know if a configuration has been delivered._ 
+
+   Solutions:
+   - Use QoS levels for message delivery confirmation.
```

Comment:
In a case of a gateway, you can't guarantee the configuration was actually applied to the endpoint using QoS levels.

Also, if you send a configuration to a client that does not have this endpoint attached, it still must respond with a puback (or die hard).